### PR TITLE
Cachito env file

### PR DESCRIPTION
**Pull request by lkolacek**
Closes CLOUDBLD-3475

Signed-off-by: Ladislav Kolacek lkolacek@redhat.com

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_download_remote_source.py [hunk 1]
```diff
@@ -63,17 +63,16 @@ def generate_cachito_env_file(self):
         cachito request.
         """
 
-        if self.buildargs:
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_download_remote_source.py:66]: Could the unit test be parametrized (with buildargs, without buildargs)?

**Review by chmeliik — COMMENTED** [review]
LGTM, though I would like a unit test if possible

File: atomic_reactor/plugins/pre_download_remote_source.py (hunk 1 above)
Comment (lkolacek) [reply atomic_reactor/plugins/pre_download_remote_source.py:66]: Unit test added. PTAL

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by rcerven — APPROVED** [review]

**Comment by ben-alkov** [comment]
@MartinBasti, @rcerven;

1. Do we need user documentation for this?
1. Is checklist item "New feature can be disabled from a configuration file" relevant?

File: atomic_reactor/plugins/pre_download_remote_source.py [hunk 2]
```diff
@@ -63,17 +63,16 @@ def generate_cachito_env_file(self):
...
-
-            # Use dedicated dir in container build workdir for cachito.env
-            abs_path = os.path.join(self.workflow.builder.df_dir,
-                                    self.REMOTE_SOURCE, CACHITO_ENV_FILENAME)
-            with open(abs_path, 'w') as f:
-                f.write('#!/bin/bash\n')
-                for env_var, value in self.buildargs.items():
-                    f.write('export {}={}\n'.format(env_var, quote(value)))
+        self.log.info('Creating %s file with environment variables '
+                      'received from cachito request', CACHITO_ENV_FILENAME)
```
Comment (tkdchen) [inline atomic_reactor/plugins/pre_download_remote_source.py:67]: > received from cachito request

Is this description correct? The `self.buildargs` is described in the `__init__` docstring that it is the container build args.

**Comment by lkolacek** [comment]
@ben-alkov added documentation change here https://github.com/containerbuildsystem/osbs-docs/pull/171
I was waiting for approvals for a new alias.

File: atomic_reactor/plugins/pre_download_remote_source.py (hunk 2 above)
Comment (lkolacek) [reply atomic_reactor/plugins/pre_download_remote_source.py:67]: I believe so. Here is a snippet of ARGs added to container build Dockefile:
* http://pastebin.test.redhat.com/923283

**Comment by MartinBasti** [comment]
@ben-alkov 
> 2. Is checklist item "New feature can be disabled from a configuration file" relevant?
It's safe to not have it, users can just ignore that file

**Review by MartinBasti — APPROVED** [review]

**Pull request merged**